File: Star Tracker/test/UnitTest.cpp
#include "UnitTest.hpp"


/**
 * @brief prints an easy to read header.
 * @param fileName The name of the file.
 * @param testFileName The name of the testFile.
 */

void printHeader ( string fileName, string testFileName )
{
	cout << endl << endl << "-------  " <<
	 	fileName << " | " << testFileName << "  ------------------" << endl;
}



/**
 * @brief Prints a readable unit test id and if passed or not.
 * @param test The function.
 * @param pass If the test passed.
 * @return pass.
 */

bool printPass ( string test, bool pass )
{
	cout << test << ":" << endl;
	if ( pass )	cout << ANSI_WHITE_BACKGROUND << ANSI_GREEN << "PASSED" << ANSI_RESET << endl;
	else		cout << ANSI_RED_BACKGROUND << ANSI_WHITE << "FAILED" << ANSI_RESET << endl;
	cout << endl;
	return pass;
}



/**
* @brief Eases comparisons.
* @param s A double.
* @param d A double.
* @return true if equal.
*/

bool equal ( decimal d1, decimal d2 )
{
	return abs(d1 - d2) < 0.001d;
}

File: Database/test/UnitTest.java

public abstract class UnitTest
{
	// Displays the test number.
	public static int testNum = 0;
	public static int testsPassed = 0;


	//Unix color codes
	public static final String ANSI_RESET = "\u001B[0m";
	public static final String ANSI_RED = "\u001B[31m";
	public static final String ANSI_GREEN = "\u001B[32m";
	public static final String ANSI_WHITE = "\u001B[37m";

	public static final String ANSI_BLACK_BACKGROUND = "\u001B[40m";
	public static final String ANSI_RED_BACKGROUND = "\u001B[41m";


	/**
	 * Runs the unit test.
	 * @return True if all the tests passed
	 */

	public static boolean run ( )
	{
		return false;
	}



	protected static void printHeader ( String className, String testName )
	{
		System.out.println("\n\n-------  " + className + " | " + testName + "  ------------------");
	}





	protected static boolean printPass ( String test, boolean pass )
	{
		System.out.println("Test: " + testNum + ", \n" + test);

		if ( pass )	System.out.println(ANSI_GREEN + "PASSED" + ANSI_RESET);
		else		System.out.println(ANSI_RED_BACKGROUND + ANSI_WHITE + "FAILED" + ANSI_RESET);
		System.out.println();
		testNum++;
		testsPassed += (pass ? 1 : 0);
		return pass;
	}



	/**
	* Eases comparisons.
	* @param s A star.
	* @param d A double.
	* @return true if equal.
	*/

	protected static boolean equal ( Star s, double d )
	{
		return Math.abs(s.attribute - d) < 0.001d;
	}






	/**
	* Eases comparisons.
	* @param s A double.
	* @param d A double.
	* @return true if equal.
	*/

	protected static boolean equal ( double d1, double d2 )
	{
		return Math.abs(d1 - d2) < 0.001d;
	}

}
